dask/dataframe/io/parquet/core.py

```diff
@@ -184,15 +185,35 @@ def read_parquet(
     read_from_paths : bool, default None
         Only used by ``ArrowDatasetEngine`` when ``filters`` are specified.
         Determines whether the engine should avoid inserting large pyarrow
-        (``ParquetFileFragment``) objects in the task graph. If this option
-        is True (default), ``read_partition`` will need to regenerate the
-        appropriate fragment object from the path and row-group IDs.  This will
-        reduce the size of the task graph, but will add minor overhead to
-        ``read_partition``.
-    chunksize : int, str
-        The target task partition size.  If set, consecutive row-groups
-        from the same file will be aggregated into the same output
-        partition until the aggregate size reaches this value.
+        (``ParquetFileFragment``) objects in the task graph.  If this option
+        is True, ``read_partition`` will need to regenerate the appropriate
+        fragment object from the path and row-group IDs.  This will reduce the
+        size of the task graph, but will add minor overhead to ``read_partition``.
+        By default (None), ``ArrowDatasetEngine`` will set this option to
+        ``False`` when there are filters.
+    chunksize : int, str, tuple, default None
+        The target output partition size. If specified, adjacent row-groups
+        and/or files will be aggregated into the same output partition until
+        the cumulative parquet-storage size reaches this value. By default,
+        inter-file aggregation will only occur if said files are stored in
+        the same directory.
+
+        Note that a non-default chunksize-aggregation range can be specified
+        by passing ``chunksize`` as a tuple with two elements. The first
+        element should be the usual int or str-based representation of the
+        desired partition size. The second element should be a string
+        specifying the aggregation range. The value "file" declares that
+        aggregation may only occur within a single file. The value "directory"
+        declares that aggregation may only occur within an immediate directory.
+        If a different string value is specified it will be used for regular
+        expression matching in the file path. For example, when reading a dataset
+        partitioned by multiple columns, like "month" and "day", we can set
+        the aggregation range to be the "day" partition by passing something
+        like the following for ``chunksize``::
+
+            ``chunksize=("1MiB", "day=[^/]*/")``
+
+        Note that the example above assumes a "/" directory separator.
```

Comment:
@randerzander - Let me know if you think this API makes sense.  Summary: The user may pass a `(size, range)` tuple for the `chunksize` parameter.  The range can just be "file" or "directory", but it can also be a general regular-expression string (to enable arbitrary directory-group aggregation).

Reply:
This API makes sense, I think.

In your day-month example, if there were additional subfolders for "hour" within each day, you should be able to specify `chunksize=("1MiB", "day=[^/]*/*") which would allow Dask to concat all the files within all the hour subfolders of a single day, correct?

Reply:
Yes, exactly - I added a test with partition names like "a" and "b" that illustrate this.

Reply:
* I think allowing for "non-default" aggregations across directories is a very useful feature!
* Is there an actual usecase for allowing regex? In my experience, most of the aggregations users would care about are partition columns / directories or "free-for-all". The regex might be very powerful but it also gives the wrong incentive imho. I don't think we should steer people into encoding information in the file paths. In my experience it is almost always easier and more performant to encode this information in the parquet files themselves
* Any reason why the "aggregation level" / second tuple element is not an argument of its own? It is not necessarily a "chunk size", is it?

For reference, we have something similar in kartothek which is called `dispatch_by` (see parameters of https://kartothek.readthedocs.io/en/latest/_rst/kartothek.io.dask.dataframe.html#kartothek.io.dask.dataframe.read_dataset_as_ddf) which concatenates entire files based on the given index columns. 

Another idea: Allow aggregations for all row groups with overlapping statistics in a given column, regardless of where the RG comes from? Out of scope for this PR but maybe something to try out / play with? idk